website/docs/tutorials/08-cairo-1-support/03-compiling.md

```diff
@@ -0,0 +1,70 @@
+---
+sidebar_label: Compilation
+---
+
+# Compiling a Cairo 1 Project with Protostar
+
+To compile your StarkNet contracts written in cairo 1 using Protostar, follow these steps:
+
+1. Specify the contracts and their corresponding Cairo source files in the [`protostar.toml` configuration file](/docs/tutorials/configuration-file).
+Each Cairo file that contains an entrypoint function should be listed in the [`contracts` section of the configuration file](/docs/tutorials/configuration-file#contracts).
+An entrypoint is a function decorated with `#[external]` or `#[view]`.
+2. Run the [`protostar build-cairo1`](../../cli-reference.md#build-cairo1) command.
+
+## Output directory
+
+Running [`protostar build-cairo1`](../../cli-reference.md#build-cairo1) command will generate the compiled contracts in the `build` directory by default.
+You can specify a custom output directory using the [`--compiled-contracts-dir`](../../cli-reference#compiled-contracts-dir-pathbuild-1) argument.
+
+```
+$ protostar build-cairo1 --compiled-contracts-dir out
+```
+
+This will create the following files in the `out` directory:
+
+```
+protostar-project
+├── src
+│   ├── main.cairo
+│   └── proxy.cairo
+├── out
+│   ├── main.json
+│   ├── proxy.json
+└── protostar.toml
+```
+
+## External libraries
+
+If you want to attach external libraries to the build, you can specify additional paths that will be searched for the cairo libraries. To do that, you need to use the [`--cairo-path`](../../cli-reference.md#cairo-path-path-1) argument like this:
+
+```
+$ protostar build-cairo1 --cairo_path /path/to/the/external/lib
+```
+
+:::info
+In order to properly create a cairo 1 library, you need to include at least three files in its directory:
```

Comment:
```suggestion
In order to properly create a Cairo 1 library, you need to include at least three files in its directory:
```

Reply:
Could we create a separate sub-tutorial for cairo1 modules? I think there is some duplication between tutorials

Reply:
> Could we create a separate sub-tutorial for cairo1 modules? I think there is some duplication between tutorials

Can we do this as a separate task?

Reply:
this was done in https://github.com/software-mansion/protostar/pull/1588